tests/graphicsItems/test_PlotDataItem.py

```diff
@@ -71,6 +72,35 @@ def test_setData():
     pdi.setData([],[])
     assert pdi.xData is None
     assert pdi.yData is None
+    
+def test_nonfinite():
+    def _assert_equal_arrays(a1, a2):
+        assert a1.shape == a2.shape
+        for ( xtest, xgood ) in zip( a1, a2 ):
+            assert( (xtest == xgood) or (np.isnan(xtest) and np.isnan(xgood) ) ) 
+        
+    x = np.array([-np.inf, 0.0, 1.0,  2.0  , np.nan,   4.0 , np.inf])
+    y = np.array([    1.0, 0.0,-1.0, np.inf,   2.0 , np.nan,   0.0 ])
+    pdi = pg.PlotDataItem(x, y)
+    dataset = pdi.getDisplayDataset()
+    _assert_equal_arrays( dataset.x, x )
+    _assert_equal_arrays( dataset.y, y )
+   
+    with warnings.catch_warnings(): 
+        warnings.simplefilter("ignore")
+        x_log = np.log10(x)
+        y_log = np.log10(y)
+    x_log[ ~np.isfinite(x_log) ] = np.nan
+    y_log[ ~np.isfinite(y_log) ] = np.nan
+
+    pdi.setLogMode(True, True)
+    dataset = pdi.getDisplayDataset()
+    print( 'good x', x_log)
+    print( 'data x', dataset.x)
+    print( 'good y', y_log)
+    print( 'data y', dataset.y)
```

Comment:
do we want the test suite printing stuff? (I know it does in other places), often pytest can suppress output, and I often have pytest _not_ suppress output when debugging.  Not saying this is a deal breaker; but just want to verify that the extra noise in the console here is worthwhile

Reply:
Next time, just say "Hey! Print statements!"   (facepalm)

No, there are there because it took me a while to figure out why arrays are never considered equal when there are any np.nan values in there:   np.nan != np.nan.

Reply:
haha well right now print statements is how we do our logging 😬  could be completely legit reasons they're there.

Reply:
--> stripped out